components/script/dom/bindings/typedarrays.rs

```diff
@@ -31,6 +31,40 @@ unsafe impl<T> crate::dom::bindings::trace::JSTraceable for HeapTypedArray<T> {
     }
 }
 
+pub fn new_initialized_heap_typed_array<T>(
+    init: HeapTypedArrayInit,
+) -> Result<HeapTypedArray<T>, ()>
+where
+    T: TypedArrayElement + TypedArrayElementCreator,
+    T::Element: Clone + Copy,
+{
+    match init {
+        HeapTypedArrayInit::Object(js_object) => Ok(HeapTypedArray {
```

Comment:
nit:

let heap_typed_array = match init { early return for errors}
Ok( heap_typed_array)

Reply:
Done

Reply:
What I meant was the below, I think it's simpler because there is only one OK
```rust
    let heap_typed_array = match init {
        HeapTypedArrayInit::Object(js_object) => HeapTypedArray {
            internal: Heap::boxed(js_object),
            phantom: PhantomData::default(),
        },
        HeapTypedArrayInit::Info { len, cx } => {
            rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
            let typed_array_result =
                create_typed_array_with_length::<T>(cx, len as usize, array.handle_mut());
            if typed_array_result.is_err() {
                return Err(());
            }
            let heap_typed_array = HeapTypedArray::<T>::default();
            heap_typed_array.internal.set(*array);
            heap_typed_array
        },
    };
    Ok(heap_typed_array)
```